Unexpected syntax errors of `a < (b >= c ...)`  in TS 4.7

# Bug Report

<!--
  Please fill in each section completely. Thank you!
-->

### 🔎 Search Terms

- Syntax error
- error TS1005
<!--
  What search terms did you use when trying to find an existing bug report?
  List them here so people in the future can find this one more easily.
-->

### 🕗 Version & Regression Information

<!-- When did you start seeing this bug occur?

"Bugs" that have existed in TS for a long time are very likely to be FAQs; refer to
  https://github.com/Microsoft/TypeScript/wiki/FAQ#common-bugs-that-arent-bugs

If possible, please try testing the nightly version of TS to see if it's already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly

Note: The TypeScript Playground can be used to try older versions of TypeScript.

Please keep and fill in the line that best applies:
-->
- This changed between versions __4.6.3__ and __4.7.2__

### ⏯ Playground Link

<!--
  A link to a TypeScript Playground "Share" link which shows this behavior

  The TypeScript Workbench can be used for more complex setups, try
  https://www.typescriptlang.org/dev/bug-workbench/

  As a last resort, you can link to a repo, but these will be slower for us to investigate.
-->
[Playground link with relevant code](https://www.typescriptlang.org/play?#code/EQVwzgpgBGAuBOBLAxrYBuAsAKGQewDs4oICQBbKAWQE8A1CeKAbygDcBGKAXig4Bp2AJh5QRAXxwAbCLHaNRQjjnxE5AQ1FsFAHigAKWg3gA6TlAB8vI4zMiA-NXq3zALifG7ASiA)

### 💻 Code

<!-- Please post the relevant code sample here as well-->
```ts
"use strict";
const enum MyVer { v1 = 1, v2 = 2 }
let ver = 21
const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
```

### 🙁 Actual behavior

<!-- What happened, and why it was wrong -->

It reports `error TS1005`:
``` log
R:/working/a.ts:4:27 - error TS1005: ')' expected.

4 const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
                            ~

R:/working/a.ts:4:60 - error TS1005: ',' expected.

4 const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
                                                             ~


Found 2 errors in the same file, starting at: R:/working/a.ts:4
```


### 🙂 Expected behavior

<!-- What you expected to happen instead, and why -->

The part between `(` and `)` should be parsed as an operand of `<`.

--- a/src/compiler/parser.ts
+++ b/src/compiler/parser.ts
@@ -5764,10 +5764,11 @@ namespace ts {
             nextToken();
 
             const typeArguments = parseDelimitedList(ParsingContext.TypeArguments, parseType);
-            if (!parseExpected(SyntaxKind.GreaterThanToken)) {
+            if (reScanGreaterToken() !== SyntaxKind.GreaterThanToken) {
                 // If it doesn't have the closing `>` then it's definitely not an type argument list.
                 return undefined;
             }
+            nextToken();
 
             // We successfully parsed a type argument list. The next token determines whether we want to
             // treat it as such. If the type argument list is followed by `(` or a template literal, as in

--- a/tests/baselines/reference/instantiationExpressionErrors.errors.txt
+++ b/tests/baselines/reference/instantiationExpressionErrors.errors.txt
@@ -141,3 +141,9 @@ tests/cases/conformance/types/typeParameters/typeArgumentLists/instantiationExpr
         protected bar = 123
     }
     
+    // Repro from #49551
+    
+    const enum MyVer { v1 = 1, v2 = 2 }
+    let ver = 21
+    const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
+    

--- a/tests/baselines/reference/instantiationExpressionErrors.js
+++ b/tests/baselines/reference/instantiationExpressionErrors.js
@@ -97,6 +97,12 @@ class C4 {
     protected bar = 123
 }
 
+// Repro from #49551
+
+const enum MyVer { v1 = 1, v2 = 2 }
+let ver = 21
+const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
+
 
 //// [instantiationExpressionErrors.js]
 "use strict";
@@ -179,6 +185,8 @@ var C4 = /** @class */ (function () {
     }
     return C4;
 }());
+var ver = 21;
+var a = ver < (1 /* MyVer.v1 */ >= 2 /* MyVer.v2 */ ? 1 /* MyVer.v1 */ : 2 /* MyVer.v2 */);
 
 
 //// [instantiationExpressionErrors.d.ts]
@@ -283,3 +291,9 @@ declare class C4 {
     };
     protected bar: number;
 }
+declare const enum MyVer {
+    v1 = 1,
+    v2 = 2
+}
+declare let ver: number;
+declare const a: boolean;

--- a/tests/baselines/reference/instantiationExpressionErrors.symbols
+++ b/tests/baselines/reference/instantiationExpressionErrors.symbols
@@ -239,3 +239,29 @@ class C4 {
 >bar : Symbol(C4.bar, Decl(instantiationExpressionErrors.ts, 94, 36))
 }
 
+// Repro from #49551
+
+const enum MyVer { v1 = 1, v2 = 2 }
+>MyVer : Symbol(MyVer, Decl(instantiationExpressionErrors.ts, 96, 1))
+>v1 : Symbol(MyVer.v1, Decl(instantiationExpressionErrors.ts, 100, 18))
+>v2 : Symbol(MyVer.v2, Decl(instantiationExpressionErrors.ts, 100, 26))
+
+let ver = 21
+>ver : Symbol(ver, Decl(instantiationExpressionErrors.ts, 101, 3))
+
+const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
+>a : Symbol(a, Decl(instantiationExpressionErrors.ts, 102, 5))
+>ver : Symbol(ver, Decl(instantiationExpressionErrors.ts, 101, 3))
+>MyVer.v1 : Symbol(MyVer.v1, Decl(instantiationExpressionErrors.ts, 100, 18))
+>MyVer : Symbol(MyVer, Decl(instantiationExpressionErrors.ts, 96, 1))
+>v1 : Symbol(MyVer.v1, Decl(instantiationExpressionErrors.ts, 100, 18))
+>MyVer.v2 : Symbol(MyVer.v2, Decl(instantiationExpressionErrors.ts, 100, 26))
+>MyVer : Symbol(MyVer, Decl(instantiationExpressionErrors.ts, 96, 1))
+>v2 : Symbol(MyVer.v2, Decl(instantiationExpressionErrors.ts, 100, 26))
+>MyVer.v1 : Symbol(MyVer.v1, Decl(instantiationExpressionErrors.ts, 100, 18))
+>MyVer : Symbol(MyVer, Decl(instantiationExpressionErrors.ts, 96, 1))
+>v1 : Symbol(MyVer.v1, Decl(instantiationExpressionErrors.ts, 100, 18))
+>MyVer.v2 : Symbol(MyVer.v2, Decl(instantiationExpressionErrors.ts, 100, 26))
+>MyVer : Symbol(MyVer, Decl(instantiationExpressionErrors.ts, 96, 1))
+>v2 : Symbol(MyVer.v2, Decl(instantiationExpressionErrors.ts, 100, 26))
+

--- a/tests/baselines/reference/instantiationExpressionErrors.types
+++ b/tests/baselines/reference/instantiationExpressionErrors.types
@@ -325,3 +325,36 @@ class C4 {
 >123 : 123
 }
 
+// Repro from #49551
+
+const enum MyVer { v1 = 1, v2 = 2 }
+>MyVer : MyVer
+>v1 : MyVer.v1
+>1 : 1
+>v2 : MyVer.v2
+>2 : 2
+
+let ver = 21
+>ver : number
+>21 : 21
+
+const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
+>a : boolean
+>ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2) : boolean
+>ver : number
+>(MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2) : MyVer
+>MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2 : MyVer
+>MyVer.v1 >= MyVer.v2 : boolean
+>MyVer.v1 : MyVer.v1
+>MyVer : typeof MyVer
+>v1 : MyVer.v1
+>MyVer.v2 : MyVer.v2
+>MyVer : typeof MyVer
+>v2 : MyVer.v2
+>MyVer.v1 : MyVer.v1
+>MyVer : typeof MyVer
+>v1 : MyVer.v1
+>MyVer.v2 : MyVer.v2
+>MyVer : typeof MyVer
+>v2 : MyVer.v2
+

--- a/tests/cases/conformance/types/typeParameters/typeArgumentLists/instantiationExpressionErrors.ts
+++ b/tests/cases/conformance/types/typeParameters/typeArgumentLists/instantiationExpressionErrors.ts
@@ -98,3 +98,9 @@ class C4 {
     protected specialFoo = f<string>
     protected bar = 123
 }
+
+// Repro from #49551
+
+const enum MyVer { v1 = 1, v2 = 2 }
+let ver = 21
+const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)

